typescript/infra/src/utils/gcloud.ts

```diff
@@ -95,6 +101,53 @@ export async function gcpSecretExists(secretName: string) {
   return matches.length > 0;
 }
 
+/**
+ * Uses the SecretManagerServiceClient to check if a secret exists.
+ * @param secretName The name of the secret to check.
+ * @returns A boolean indicating whether the secret exists.
+ */
+export async function gcpSecretExistsUsingClient(
+  secretName: string,
+  client?: SecretManagerServiceClient,
+): Promise<boolean> {
+  if (!client) {
+    client = new SecretManagerServiceClient({
```

Comment:
nit: just noticing that we do this in various places, maybe a candidate for a `getSecretManagerClient` fn or something